addons/mail/static/src/components/channel_invitation_form/channel_invitation_form.xml

```diff
@@ -69,6 +69,10 @@
                                 </button>
                             </div>
                         </div>
+
+                        <t t-if="channelInvitationForm.thread.authorizedGroup and channelInvitationForm.thread.public == 'groups'">
+                            Access restricted to group "<span t-esc="channelInvitationForm.thread.authorizedGroup"></span>"
```

Comment:
wrap the whole sentence in some kind of tag too (at least span, possibly div?)

As for the "text injection" prefer using %s in a translated string to ease the translation (no tag inside, easier to move the words around)